script/utils/update-redirects-data.ts

```diff
@@ -60,10 +75,20 @@ const updateRedirectsData = async(currentBook: BookWithOSWebData, newBook: BookW
       countNewRedirections++;
     // remove `else` to enable legitimately removing pages from books
     // only once uuids are guaranteed to be consistent
-    } else if (!newSection && matchSlug(section.slug) === undefined && !matchException) {
+    } else if (!newSection && matchSlug(section.slug) === undefined && !matchException && !booksAreDifferent) {
       throw new Error(
         `updateRedirects prohibits removing pages from a book, `
         + `but neither section with ID ${section.id} nor slug ${section.slug} was found in book ${newBook.id}`);
+    } else if (booksAreDifferent && !redirects.find(matchRedirect(section))) {
```

Comment:
lets do the matchException check here too, like if we didn't throw on the previous condition because of that then also don't go in this logic